### PR TITLE
feat(input): Add size prop

### PR DESCRIPTION
**Pull request by vuluongj20**
Add support for a `size` prop on `Input`, to prevent inconsistencies around input sizing, and ensure that inputs and buttons have the same height & similar padding.

image

### CODE REVIEW

File: static/app/components/input.tsx [hunk 1]
```diff
@@ -1,21 +1,87 @@
...
+        // native `size` attribute (for that, use `nativeSize` instead)
+        size: _size,
+        // Use `nativeSize` as the native `size` attribute
+        nativeSize,
+        ...props
+      },
+      ref
+    ) => <input {...props} ref={ref} size={nativeSize} />
+  ),
+  {shouldForwardProp: prop => typeof prop === 'string' && isPropValid(prop)}
```
Comment (evanpurkhiser) [inline static/app/components/input.tsx:82]: you don't need this anymore, the default is to check `isPropValid`

Comment (vuluongj20) [reply static/app/components/input.tsx:82]: I tried deleting it but a bunch of tests (~200) failed with this error message:
image

Comment (evanpurkhiser) [reply static/app/components/input.tsx:82]: oic 🤔

File: static/app/components/input.tsx [hunk 2]
```diff
@@ -1,21 +1,87 @@
...
   type?: React.HTMLInputTypeAttribute;
 }
 
-const Input = styled('input', {
-  shouldForwardProp: prop =>
-    // Do not forward required to `input` to avoid default browser behavior
-    typeof prop === 'string' && isPropValid(prop) && prop !== 'required',
-})<InputProps>`
-  ${inputStyles};
+export const inputStyles = (p: InputProps & {theme: Theme}) => css`
```
Comment (malwilley) [inline static/app/components/input.tsx:16]: So it looks like we've got `inputStyles` being exported from two places now - here, and in `sentry/styles/input`. Are we okay with having both exist?

Comment (vuluongj20) [reply static/app/components/input.tsx:16]: Nope! The `inputStyles` exported here will the the main one. My plan is to go through all the `sentry/styles/input` imports and either replace them with this `<Input />` component, or if that's not possible redirect the import path to here (`sentry/components/input`). After all that is done, I'll delete `styles/input`. This PR is the first step in that process.

Comment (malwilley) [reply static/app/components/input.tsx:16]: Sounds good! 👍

**Review by malwilley — APPROVED** [review]

**Pull request merged**